File: wlr/SPX.States/StateManager.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SPX.States
{
    public class StateManager
    {
        private static State m_state;

        public static void LoadState(State state)
        {
            m_state = state;
            m_state.LoadContent();
        }

        public static void Draw()
        {
            m_state.Draw();
        }

        public static void LoadContent()
        {
            m_state.LoadContent();
        }

        public static void Update()
        {   
            m_state.Update();
        }
    }
}

File: aigilas/src/sps/states/StateManager.java
package sps.states;

import sps.text.TextPool;

public class StateManager {
    private static State _state;

    public static void loadState(State state) {
        if (_state != null) {
            _state.unload();
        }
        TextPool.get().clear();
        _state = state;
        _state.load();
    }

    public static void draw() {
        _state.draw();
    }

    public static void loadContent() {
        _state.load();
    }

    public static void update() {
        _state.update();
    }

    public static void asyncUpdate() {
        _state.asyncUpdate();
    }
}
